src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,28 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 100;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null
+                || Storage.people.contains(storageDao.get(user.getLogin()))
+                || user.getAge() < MIN_AGE
+                || user.getAge() > MAX_AGE
+                || user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RuntimeException();
+        }
```

Comment:
you should split this if into several small ones and pass appropriate informative messages to the constructors